serverlessworkflow/modules/ROOT/pages/testing-and-troubleshooting/mocking-http-cloudevents-with-wiremock.adoc

```diff
@@ -1 +1,194 @@
+     * Starts the "sink" server, which is is the endpoint that will receive our produced events
+     */
+    @BeforeAll <5>
+    public static void startSink() {
+        sink = new WireMockServer(options().port(8181)); <6>
+        sink.start(); <7>
+        sink.stubFor(post("/").willReturn(aResponse().withBody("ok").withStatus(200))); <8>
+    }
+
+    @AfterAll <9>
+    public static void stopSink() {
+        if (sink != null) {
+            sink.stop(); <10>
+        }
+    }
+
+    @Test
+    void processDomesticOrderUnderFraudEval() throws JsonProcessingException, InterruptedException {
+        final ObjectMapper objectMapper = new ObjectMapper();
+        final Order order = new Order();
+        order.setId(UUID.randomUUID().toString());
+        order.setDescription("iPhone 12");
+        order.setTotal(1001);
+        order.setCountry("US");
+
+        given() <4>
+                .header("ce-specversion", "1.0")
+                .header("ce-id", order.getId())
+                .header("ce-source", "/from/test")
+                .header("ce-type", "orderEvent")
+                .contentType(MediaType.APPLICATION_JSON)
+                .body(objectMapper.writeValueAsString(order))
+                .post("/")
+                .then()
+                .statusCode(200);
+
+        await() <11>
+                .atMost(60, SECONDS)
+                .with().pollInterval(1, SECONDS)
+                .untilAsserted(() -> sink.verify (2, postRequestedFor(urlEqualTo("/")).withRequestBody(containing(order.getId())))); <12>
+    }
+}
+
+----
+<1> @QuarkusTest starts a Quarkus server for the whole lifetime of the tests execution run.
+More details in {recap_of_http_based_testing_in_jvm_mode}[enable quarkus testing resources].
+<2> The naming of the test would end with 'IT' to identify which test needs to be executed as an integration test.
+<3> {wiremockserver_class}[WireMockServer] mocked server instance used for Sink Binding for testing.
+<4> Used to test interactions with the application, more explained in the guide {basic_integration_test_with_restassured_guide}[Testing your Serverless Workflow application using REST Assured].
+<5> `@BeforeAll` annotation used to signal that the annotated method should be executed before all the tests run.
+<6> Creates the {wiremockserver_class}[WireMockServer] instance listening at the port passed as parameter and must match with the Sink configuration.
+<7> Start the server before the tests executed.
+<8> Stub the mock API response.It accepts a MappingBuilder instance that we can use to build API mapping information such as URL, request parameters and body, headers, authorization etc.
+<9> `@AfterAll` annotation used to signal that the annotated method should be executed after all the tests execution.
+<10> Stop the server after all test execution.
+<11> {awaitility_class}[`await()`] added to waiting for asynchronous operations.
+<12> {wiremock_verifying}[`verify`] if the request has hit the mock API.
+
```

Comment:
Do you mind explaining what happens when we run this test? For example, the `await` part is super important, since it might shed a light into users' heads. We are waiting for the sink to receive the cloud event produced by the workflow. I think images would be nice here to illustrate it.